debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/SnapshotReader.java

```diff
@@ -866,8 +866,16 @@ private void readTableSchema(final AtomicReference<String> sql, final JdbcConnec
         });
     }
 
-    private boolean shouldRecordTableSchema(final MySqlSchema schema, final Filters filters, TableId id) {
-        return !schema.isStoreOnlyMonitoredTablesDdl() || filters.tableFilter().test(id);
+    /**
+     * Whether DDL for the given table should be recorded.
+     */
+    private boolean shouldRecordTableSchema(MySqlSchema schema, Filters filters, TableId id) {
+        // filtered, built-in, or globally ignored table
+        if (!filters.tableFilter().test(id)) {
+            return false;
+        }
+
+        return !schema.isStoreOnlyMonitoredTablesDdl();
```

Comment:
So this would solve the "phantom" tables issue but will unfortunately cause a regression:

**config:**
```
database.history.store.only.monitored.tables.ddl=false
table.whitelist=mydb.table1
```

**input:**
```
id: mydb.table2
```

**Predicate:**
```
if (!filters.tableFilter().test(id)) {
   return false;
}
```

Will then return `false` which is not what we want for this configuration.

Reply:
Yeah :( Seems there's indeed no way around querying the ignore predicate separately.

Reply:
Not unless we somehow refactor the `database.history.store.only.monitored.tables.ddl` to be visible in the `Filters` class in order to take it into account internally? But I'm not sure what's LOE there.

Reply:
Yeah, in any case not worth the effort at this point. There's larger refactorings ahead for the MySQL connector in general and filter handling in particular.